modules/stochastic_tools/test/src/vectorpostprocessors/PODSurrogateTester.C

```diff
@@ -0,0 +1,105 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// Stocastic Tools Includes
+#include "PODSurrogateTester.h"
+
+#include "Sampler.h"
+
+registerMooseObject("StochasticToolsTestApp", PODSurrogateTester);
+
+InputParameters
+PODSurrogateTester::validParams()
+{
+  InputParameters params = GeneralVectorPostprocessor::validParams();
+  params += SamplerInterface::validParams();
+  params += SurrogateModelInterface::validParams();
+  params.addClassDescription("Tool for sampling POD surrogate model.");
+  params.addRequiredParam<UserObjectName>("model", "Name of surrogate model.");
+  params += SamplerInterface::validParams();
+  params.addRequiredParam<SamplerName>(
+      "sampler", "Sampler to use for evaluating POD model (mainly for testing).");
+  params.addParam<bool>("output_samples",
+                        false,
+                        "True to output value of samples from sampler (this may be VERY large).");
+  params.addRequiredParam<std::string>(
+      "variable_name", "The name of the variable this prostprocessor is supposed to operate on.");
+  MooseEnum pptype("nodal_max=0 nodal_min=1 nodal_l1=2 nodal_l2=3 nodal_linf=4");
+  params.addRequiredParam<MooseEnum>(
+      "to_compute", pptype, "The global data the postprocessor should compute.");
+  return params;
+}
+
+PODSurrogateTester::PODSurrogateTester(const InputParameters & parameters)
+  : GeneralVectorPostprocessor(parameters),
```

Comment:
Could you derive this from EvaluateSurrogate? It will be a couple extra input params, dynamic casting the surrogates, and overriding the execute function.

Reply:
For POD-RB surrogates, it is more efficient to get more QoIs for one surrogate, and that makes this inheritance complicated since a postprocessor value with the name of the model is always registered in the base class. Thus, I just refurnished this class but didn't inherit from EvaluateSurrogate.